### PR TITLE
[All Hosts] (ui) Code snippets for Office.Tab and Office.UI.

### PR DESCRIPTION
**Pull request by Rick-Kirkham**

### CODE REVIEW

File: docs/code-snippets/office-snippets.yaml [hunk 1]
```diff
@@ -3356,6 +3356,52 @@ Office.Settings#saveAsync:member(2):
...
+                    visible: true
+                }
+            ]});
+    }
+
+    // The the following does the same thing in TypeScript.
+
+    const showDataTab = async () => {
+        const myContextualTab: Office.Tab = {id: "CtxTab1", visible: true};
+        const ribbonUpdater: Office.RibbonUpdaterData = { tabs: [ myContextualTab ]};
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3379]:
```suggestion
        const ribbonUpdater: Office.RibbonUpdaterData = { tabs: [ myContextualTab ] };
```

File: docs/code-snippets/office-snippets.yaml [hunk 2]
```diff
@@ -3499,6 +3545,32 @@ Office.TableData#rows:member:
         tableData.rows = [['row1'], ['row2'], ['row3']];
         return tableData;
     }
+Office.UI#displayDialogAsync:member(1):
+  - |-
+    // The following example shows how to open a dialog with a specified size. It also shows
+    // how to register a function to handle the message when Office.UI.messageParent() is called
+    // in the dialog. The implementation of the processMessage() function is omitted.
+
+    Office.context.ui.displayDialogAsync('https://www.contoso.com/myDialog.html', {height: 30, width: 20},
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3554]:
```suggestion
    Office.context.ui.displayDialogAsync("https://www.contoso.com/myDialog.html", { height: 30, width: 20 },
```

File: docs/code-snippets/office-snippets.yaml [hunk 3]
```diff
@@ -3356,6 +3356,52 @@ Office.Settings#saveAsync:member(2):
...
+                    id: "CtxTab1",
+                    visible: true
+                }
+            ]});
+    }
+
+    // The the following does the same thing in TypeScript.
+
+    const showDataTab = async () => {
+        const myContextualTab: Office.Tab = {id: "CtxTab1", visible: true};
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3378]:
```suggestion
        const myContextualTab: Office.Tab = { id: "CtxTab1", visible: true };
```
To be consistent with other code.

File: docs/code-snippets/office-snippets.yaml [hunk 4]
```diff
@@ -3356,6 +3356,52 @@ Office.Settings#saveAsync:member(2):
...
+                    id: "CtxTab1",
+                    visible: true
+                }
+            ]});
+    }
+
+    // The following does the same thing in TypeScript.
+
+    const showDataTab = async () => {
+        const myContextualTab: Office.Tab = {id: "CtxTab1", visible: true};
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3401]:
```suggestion
        const myContextualTab: Office.Tab = { id: "CtxTab1", visible: true };
```

File: docs/code-snippets/office-snippets.yaml [hunk 5]
```diff
@@ -3356,6 +3356,52 @@ Office.Settings#saveAsync:member(2):
...
+        const ribbonUpdater: Office.RibbonUpdaterData = { tabs: [ myContextualTab ]};
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3402]:
```suggestion
        const ribbonUpdater: Office.RibbonUpdaterData = { tabs: [ myContextualTab ] };
```

File: docs/code-snippets/office-snippets.yaml [hunk 6]
```diff
@@ -3499,6 +3545,32 @@ Office.TableData#rows:member:
...
         return tableData;
     }
+Office.UI#displayDialogAsync:member(1):
+  - |-
+    // The following example shows how to open a dialog with a specified size. It also shows
+    // how to register a function to handle the message when Office.UI.messageParent() is called
+    // in the dialog. The implementation of the processMessage() function is omitted.
+
+    Office.context.ui.displayDialogAsync('https://www.contoso.com/myDialog.html', {height: 30, width: 20},
+        function (asyncResult) {
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3555]:
```suggestion
        (asyncResult) => {
```
To be consistent with format used in succeeding code.

File: docs/code-snippets/office-snippets.yaml [hunk 7]
```diff
@@ -3356,6 +3356,52 @@ Office.Settings#saveAsync:member(2):
     function write(message){
         document.getElementById('message').innerText += message; 
     }
+Office.Tab#id:member:
```
Comment (samantharamon) [inline docs/code-snippets/office-snippets.yaml:3359]: Doesn't need to be done in this PR, but consider also adding the snippet to [Office.ribbon.requestUpdate](https://learn.microsoft.com/javascript/api/office/office.ribbon#office-office-ribbon-requestupdate-member(1)) and [Office.RibbonUpdaterData](https://learn.microsoft.com/javascript/api/office/office.ribbonupdaterdata).

**Review by samantharamon — APPROVED** [review]
Looks good! Added minor suggestions for your consideration.

File: docs/code-snippets/office-snippets.yaml (hunk 7 above)
Comment (Rick-Kirkham) [reply docs/code-snippets/office-snippets.yaml:3513]: > Doesn't need to be done in this PR, but consider also adding the snippet to [Office.ribbon.requestUpdate](https://learn.microsoft.com/javascript/api/office/office.ribbon#office-office-ribbon-requestupdate-member(1)) and [Office.RibbonUpdaterData](https://learn.microsoft.com/javascript/api/office/office.ribbonupdaterdata).

Done

**Pull request merged**